src/middleware/basic.js

```diff
@@ -0,0 +1,24 @@
+'use strict'
+
+const base64 = require('base-64')
+const { users } = require('../models/modelsIndex.js')
+
+module.exports = async (req, res, next) => {
+  if (!req.headers.authorization) {
+    return _authError()
+  }
+
+  let basic = req.headers.authorization.split(' ').pop()
+  let [user, pass] = base64.decode(basic).split(':')
+
+  try {
+    req.user = await users.authenticateBasic(user, pass)
+    next()
+  } catch (e) {
+    _authError()
+  }
+
+  function _authError() {
```

Comment:
lets rename this function to remove the `_` as this doesn't appear to be recursive. If it's use commonly we can keep it, but haven't seen this before.